README.md

```diff
@@ -1,2 +1,92 @@
-# container-installer
-Creates an ISO for installing a container image as an OS
+# UBlueOS ISO Generator
+This action is used to generate the ISO images for UBlueOS.
+
+
+## Makefile
+A Makefile is provided for ease of use. There are separate targets for each file generated, however `make` can be used to generate the final image and `make clean` can be used to clean up the workspace.
+
+See [Customizing](#customizing) for information about customizing the image that gets created.
+
+## Container
+A container with the necessary tools already installed is provided at `ghcr.io/ublue-os/isogenerator:latest`
+
+To use the container file, run `docker --privileged --volume .:/isogenerator/output ghcr.io/ublue-os/isogenerator`
+
+See [Customizing](#customizing) for information about customizing the image that gets created. The variable can either be defined as environment variables or as command arguments.
+Examples:
+Environment Variable
+```bash
+docker --privileged --volume .:/isogenerator/output -e VERSION=39 ghcr.io/ublue-os/isogenerator
+```
+Command Argument
+```bash
+docker --privileged --volume .:/isogenerator/output ghcr.io/ublue-os/isogenerator VERSION=39
+```
+
+## Customizing
+The following variables can be used to customize the create image.
+
+- ARCH  
+    Architecture for image to build  
+    Default Value: x86_64
+- VERSION  
+    Fedora version of installer to build  
+    Default Value: 39
+- IMAGE_REPO  
+    Repository containing the source container image  
+    Default Value: ghcr.io/ublue-os
+- IMAGE_NAME  
+    Name of the source container image  
+    Default Value: base-main
+- IMAGE_TAG  
+    Tag of the source container image  
+    Default Value: *VERSION*
+- VARIANT  
```

Comment:
Could we provide a larger list of Variants that are available? Someone wanted to build a CoreOS ISO using this tool and it would be good to know what are the valid options (or just a link to them)

Reply:
How's this for the description of VARIANT:
Source container variant. Available options can be found by running `dnf provides system-release`. Variant will be the third item in the package name. Example: `fedora-release-kinoite-39-34.noarch` will be kinonite

Also, that issue was because they were trying to put part of the repo path in the image name. Image name is not expecting to include part of the path for the repo. That's why there's a var for the repo.